docs/installation/server-config/avro-schema.rst

```diff
@@ -1,4 +1,100 @@
 .. _install_ksql-avro-schema:
 
-Avro and Confluent Schema Registry
-==================================
\ No newline at end of file
+Avro and Schema Registry
+========================
+
+KSQL can read and write messages in Avro format by integrating with :ref:`Confluent Schema Registry <schemaregistry_intro>`.
+KSQL automatically retrieves (read) and registers (write) Avro schemas as needed and thus saves you from both having to
+manually define columns and data types in KSQL and from manual interaction with the Schema Registry.
+
+.. contents:: Contents
+    :local:
+
+Supported functionality
+^^^^^^^^^^^^^^^^^^^^^^^
+
+KSQL currently supports Avro data in the Kafka message values.
+
+The following functionality is not supported yet:
+
+-  Message keys in Avro format are not supported. Message keys in KSQL are always interpreted as STRING format, which means
+   KSQL will ignore Avro schemas that have been registered for message keys.
+-  Avro schemas with nested fields are not supported yet. This is because KSQL does not yet support nested columns.
+
+Configuring KSQL for Avro
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You must configure the API endpoint of |sr| by setting ``ksql.schema.registry.url`` (default: ``http://localhost:8081``)
+in the KSQL configuration file (``<path-to-confluent>/etc/ksql/ksql-server.properties``) that you use to start the KSQL
+Server. For more information, see :ref:`install_ksql-server`.
+
+.. important:: Do not use the SET statement in the KSQL CLI to configure the registry endpoint.
+
+Using Avro in KSQL
+^^^^^^^^^^^^^^^^^^
+
+Before using Avro in KSQL, make sure that |sr| is up and running and that ``ksql.schema.registry.url`` is set correctly
+in the KSQL properties file. |sr| is :ref:`installed by default <quickstart>` with |cp|.
+
+- Read from Kafka topics with Avro-formatted data by using ``CREATE STREAM`` and ``CREATE TABLE`` statements.
+- Write Avro-formatted data into Kafka topics by using ``CREATE STREAM AS SELECT`` and ``CREATE TABLE AS SELECT`` statements.
+- Create derived streams and tables from existing streams and tables with ``CREATE STREAM AS SELECT`` and
+  ``CREATE TABLE AS SELECT`` statements.
+- Convert data to different formats with ``CREATE STREAM AS SELECT`` and ``CREATE TABLE AS SELECT`` statements. For example,
+  you can convert a stream from Avro to JSON.
+
+Example: Create a new stream ``pageviews`` by reading from a Kafka topic with Avro-formatted messages.
+    .. code:: sql
+
+        CREATE STREAM pageviews
+          WITH (KAFKA_TOPIC='pageviews-avro-topic',
+                VALUE_FORMAT='AVRO');
+
+Example: Create a new table ``users`` by reading from a Kafka topic with Avro-formatted messages.
+    In this example you don’t need to define any columns or data types in the CREATE statement because KSQL automatically
```

Comment:
When you create a new stream by reading from a Kafka topic with Avro-formatted messages, you don't have to define columns or data types in the CREATE statement....if you are interacting via the CLI.  If you are putting these commands in a script that you then pass in via `queries-file`, then I think you do.  cc: @hjafarpour

Reply:
It should work in both cases. @ybyzek can you test it with `queries-file`?